extension/entitystore/extension.go

```diff
@@ -0,0 +1,254 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+	"github.com/jellydator/ttlcache/v3"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.uber.org/zap"
+
+	configaws "github.com/aws/amazon-cloudwatch-agent/cfg/aws"
+	"github.com/aws/amazon-cloudwatch-agent/internal/ec2metadataprovider"
+	"github.com/aws/amazon-cloudwatch-agent/internal/retryer"
+	"github.com/aws/amazon-cloudwatch-agent/sdk/service/cloudwatchlogs"
+	"github.com/aws/amazon-cloudwatch-agent/translator/config"
+)
+
+const (
+	Service                     = "Service"
+	InstanceIDKey               = "EC2.InstanceId"
+	ASGKey                      = "EC2.AutoScalingGroup"
+	ServiceNameSourceKey        = "AWS.ServiceNameSource"
+	PlatformType                = "PlatformType"
+	EC2PlatForm                 = "AWS::EC2"
+	Type                        = "Type"
+	Name                        = "Name"
+	Environment                 = "Environment"
+	podTerminationCheckInterval = 5 * time.Minute
+)
+
+type ec2ProviderType func(string, *configaws.CredentialConfig) ec2iface.EC2API
+
+type serviceProviderInterface interface {
+	startServiceProvider()
+	addEntryForLogFile(LogFileGlob, ServiceAttribute)
+	addEntryForLogGroup(LogGroupName, ServiceAttribute)
+	logFileServiceAttribute(LogFileGlob, LogGroupName) ServiceAttribute
+	getServiceNameAndSource() (string, string)
+}
+
+type EntityStore struct {
+	logger *zap.Logger
+	config *Config
+	done   chan struct{}
+
+	// mode should be EC2, ECS, EKS, and K8S
+	mode string
```

Comment:
nit : The mode is set with context.CurrentContext().Mode(). So this will only have values "ec2", "onPrem" and "onPremise"